### PR TITLE
[FEA] add libcugraphops as a dependency of cugraph

### PR DESCRIPTION
**Pull request by MatthiasKohl**
We need to do several things:
- Add libcugraphops to conda (both in the cugraph recipe as well as build environment)
- Add libcugraphops to cmake (I added a new thirdparty include here)
- Ensure that the right headers and objects are added in cmake

<s>For now, this is a work in progress since libcugraphops doesn't seem available on anaconda yet.</s>

### CODE REVIEW

**Review by BradReesWork — APPROVED** [review]

**Comment by ajschmidt8** [comment]
Please consider using GitHub's _Draft PR_ feature instead of `WIP` tags in the future. Draft PRs have the benefit of preventing notifications to `codeowners` until PRs are marked _Ready for Review_, which helps cut down on excessive notifications for PRs that are still being worked on. CI will still run on Draft PRs.

Some useful information about Draft PRs:

- https://github.blog/2019-02-14-introducing-draft-pull-requests/
- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests
- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request

**Comment by MatthiasKohl** [comment]
@ajschmidt8 sorry about this, I wasn't aware of the feature, I'll try to use it from now on

**Comment by BradReesWork** [comment]
rerun tests

**Comment by BradReesWork** [comment]
rerun tests

**Review by robertmaynard — APPROVED** [review]
CMake changes LGTM

**Review by BradReesWork — APPROVED** [review]

**Review by Ethyling — CHANGES_REQUESTED** [review]

File: conda/environments/cugraph_dev_cuda11.0.yml [hunk 1]
```diff
@@ -3,9 +3,11 @@ channels:
 - rapidsai
 - nvidia
 - rapidsai-nightly
+- rapidsai-nightly/label/testing
```
[the remark this answers is not in the document]
Comment (MatthiasKohl) [reply conda/environments/cugraph_dev_cuda11.0.yml]: done.

**Comment by ChuckHastings** [comment]
rerun tests

**Comment by ChuckHastings** [comment]
rerun tests

**Comment by MatthiasKohl** [comment]
rerun tests

**Review by rlratzel — APPROVED** [review]

**Review by Ethyling — APPROVED** [review]

**Comment by rlratzel** [comment]
@gpucibot merge

**Pull request merged**